### PR TITLE
feat: Create full RFQm health check implementation

### PR DESCRIPTION
**Pull request by rhinodavid**
# Description

Here's the big one. Takes all the pieces of the RFQm health check built over the past weeks and combines them. In the handler, the result of the health check is cached to save computation resources. The user can override the caching with a `Cache-Control` header in the request.

Closes MKR-48

# Testing Instructions

Since all the individual pieces of the health check are tested, I elected to test this PR manually. With a local dev server attached to the debugger I verified the caching features and sanity checked that the `computeHealthCheckAsync` method was being called as expected.

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/utils/rfqm_health_check.ts [hunk 1]
```diff
@@ -65,6 +98,20 @@ export function transformResultToShortResponse(result: HealthCheckResult): RfqmH
     };
 }
 
+/**
+ * Changes the set of trading pairs from the format used in config to the format used in the health check response.
+ */
+function transformPairs(offerings: RfqMakerAssetOfferings): { [pair: string]: HealthCheckStatus } {
```
Comment (rhinodavid) [inline src/utils/rfqm_health_check.ts:104]: moved from the old `runHealthCheckAsync` logic

**Review by PirosB3 — APPROVED** [review]
As discussed offline, feel free to remove the cache if you feel it's worth removing. Aside from that, this is ready to go in! Feel free to merge

File: src/handlers/rfqm_handlers.ts [hunk 2]
```diff
@@ -137,7 +143,42 @@ export class RfqmHandlers {
     }
 
     public async getHealthAsync(req: express.Request, res: express.Response): Promise<void> {
-        const result = await this._rfqmService.runHealthCheckAsync();
+        // The maximum age of the cache as specified in the request
+        let cacheAgeHeaderMaxAgeMs = Infinity;
+        // Users can add the `Cache Control` header to the request to get fresh data if they need it (mostly for dev testing).
+        // The header has the form of `Cache-Control: max-age=<seconds>`. See:
+        // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#cache_request_directives
+        const cacheControlHeaderString = req.get('Cache-Control');
```
[the remark this answers is not in the document]
Comment (rhinodavid) [reply src/handlers/rfqm_handlers.ts]: disregard we're killing this

File: src/handlers/rfqm_handlers.ts [hunk 3]
```diff
@@ -75,7 +75,13 @@ const RFQM_SIGNED_QUOTE_SUBMITTED = new Counter({
     labelNames: ['apiKey'],
 });
 
+// If the cache is more seconds old than the value specified here, it will be refreshed.
+const HEALTH_CHECK_RESULT_CACHE_DURATION_S = 60;
```
[the remark this answers is not in the document]
Comment (rhinodavid) [reply src/handlers/rfqm_handlers.ts]: done

**Pull request merged**